retrofit/src/main/java/retrofit2/RequestBuilder.java

```diff
@@ -107,19 +106,16 @@ private static String canonicalize(String input, boolean alreadyEncoded) {
     return input;
   }
 
-  private static void canonicalize(Buffer out, String input, int pos, int limit,
-      boolean alreadyEncoded) {
+  private static void canonicalizeForPath(Buffer out, String input, int pos, int limit,
+                                          boolean alreadyEncoded) {
     Buffer utf8Buffer = null; // Lazily allocated.
     int codePoint;
     for (int i = pos; i < limit; i += Character.charCount(codePoint)) {
       codePoint = input.codePointAt(i);
       if (alreadyEncoded
-          && (codePoint == '\t' || codePoint == '\n' || codePoint == '\f' || codePoint == '\r')) {
+          && shouldSkipCodePoint(codePoint)) {
         // Skip this character.
-      } else if (codePoint < 0x20
-          || codePoint >= 0x7f
-          || PATH_SEGMENT_ENCODE_SET.indexOf(codePoint) != -1
-          || (codePoint == '%' && !alreadyEncoded)) {
+      } else if (!alreadyEncoded && shouldEncodeCodePoint(codePoint)) {
```

Comment:
I'm personally fine with the change of simply not canonicalizing `/` in path replacements. It makes sense in OkHttp's HttpUrl since its APIs explicitly deal with path segments. Retrofit deals with entire paths as a single unit, and the case where someone wants to dynamically determine sections of the path which may span multiple segments is not as uncommon as I would like.